### PR TITLE
Update to 0.3.7

### PR DESCRIPTION
**Pull request by t2d**
* Check sha256 sum during download

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,14 @@ Linux machine.
 Role Variables
 --------------
 
-goss_version: Defaults to v0.3.6, which is latest at the time of writing. Any new versions need to be handjammed into vars/main.yml because of the manual checksum validation. Currently supports all known versions
+    goss_version: "v0.3.7"
+    goss_sha256sum: 357f5c7f2e7949b412bce44349cd32ab19eb3947255a8ac805f884cc2c326059.
```
Comment (bbaassssiiee) [inline README.md]: The checksum should go in the list in vars/main.yml

File: defaults/main.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 # defaults file for install_goss
-goss_version: "v0.3.6"
+goss_version: "v0.3.7"
+goss_sha256sum: 357f5c7f2e7949b412bce44349cd32ab19eb3947255a8ac805f884cc2c326059.
```
Comment (bbaassssiiee) [inline defaults/main.yml]: The checksum should go in the list in vars/main.yml

File: tasks/main.yml [hunk 3]
```diff
@@ -15,6 +15,7 @@
   delegate_to: localhost
   get_url:
     url: "https://github.com/aelsabbahy/goss/releases/download/{{ goss_version }}/goss-linux-amd64"
+    sha256sum: "{{ goss_sha256sum }}"
```
Comment (bbaassssiiee) [inline tasks/main.yml]: I like the idea of checking the sha256sum, but the implementation should use the list in vars/main.yml with the goss_version provided in the group_vars of the user with your new version from defaults/main.yml if the user does not care. To implement this, this line should read:

```yaml
sha256sum: "{{ goss[goss_version].sha256sum }}"
```

File: README.md (hunk 1 above)
Comment (bbaassssiiee) [reply README.md]: The checksum validation will be automated in 3 places.

File: defaults/main.yml (hunk 2 above)
Comment (bbaassssiiee) [reply defaults/main.yml]: The new version is fine.

**Comment by t2d** [comment]
Thank you for your review. That makes a lot of sense. Do you think it's better now?

File: molecule/default/verify.yml [hunk 4]
```diff
@@ -7,10 +7,10 @@
   hosts: all
   become: true
   vars:
-    goss_version: v0.3.6
+    goss_version: v0.3.7
     goss_arch: amd64
     goss_dst: /usr/bin/goss
-    goss_sha256sum: 53dd1156ab66f2c4275fd847372e6329d895cfb2f0bcbec5f86c1c4df7236dde
+    goss_sha256sum: 357f5c7f2e7949b412bce44349cd32ab19eb3947255a8ac805f884cc2c326059.
```
Comment (bbaassssiiee) [inline molecule/default/verify.yml]: There is a dot at the end of the checksum

**Pull request merged**